auth/tasks.py

```diff
@@ -50,51 +61,79 @@ def update_group_and_perms(department, created):
         if created:
             assign_model_perms_for_department(department)
 
+    def find_all_child_department(super_department):
```

Comment:
Although our current logic indicates that no teachers are related to non-leaf departments, returning all descendants and removing users from these departments harms no one, and gives our system capabilities to support this feature.